API/Services/Implementations/PlayerStatsService.cs

```diff
@@ -260,35 +260,43 @@ DateTime dateMax
         var matchStats = (await matchStatsRepository.GetForPlayerAsync(id, ruleset, dateMin, dateMax)).ToList();
         var adjustments =
             (await ratingStatsRepository.GetForPlayerAsync(id, ruleset, dateMin, dateMax))
-            .ToList()
-            .SelectMany(x => x)
+            .GroupBy(ra => ra.Timestamp)
+            .SelectMany(ra => ra)
             .ToList();
 
-        if (matchStats.Count == 0)
+        if (matchStats.Count == 0 || adjustments.Count == 0)
         {
             return new AggregatePlayerMatchStatsDTO();
         }
 
+        var offset = adjustments.FirstOrDefault(ra => ra.AdjustmentType == RatingAdjustmentType.Initial)?.RatingAfter;
+
+        if (offset is null)
+        {
+            logger.LogWarning("Initial rating adjustment not found for player, stats will not be completely accurate " +
+                            " [Id: {PlayerId}]", id);
+        }
+
```

Comment:
I think I understand what this is doing, but can you explain why its necessary?

Reply:
Basically if you sum all of the `RatingAfter` values without subtracting from the initial rating you end up with a large negative value.

Will modify this implementation slightly because offset isn't actually required. If a user selects a timespan of say 30 days there won't be an initial rating. But, if the initial rating is included in the result set, we need to account for it.